benchopt/base.py

```diff
@@ -517,3 +518,45 @@ def __reduce__(self):
             self._module_filename, module_hash, self._parameters, dataset,
             str(self._import_ctx._benchmark_dir)
         )
+
+    def default_split(self, *arrays):
+        i_train, i_test = next(self._cv)
+        res = ()
+        for x in arrays:
+            res = (*res, x[i_train], x[i_test])
+        return res
+
+    def custom_split(self, *arrays):
+        if not hasattr(self, "_cv"):
+            # this define _cv only once
+            self._cv = itertools.cycle(self.cv.split(*arrays))
+        i_train, i_test = next(self._cv)
+        return self.split(i_train, i_test)
+
+    def get_split(self, *arrays):
+        # return the split of the data according to the cv attribute
+
+        if not hasattr(self, "cv"):
+            raise ValueError(
+                "To use get_split, you need to define "
+                "a cv attribute in your objective class "
+                "It should be a sklearn.model_selection.BaseCrossValidator "
+                "object."
+            )
+
+        if not str(type(arrays[0])) == "<class 'numpy.ndarray'>":
+            if not hasattr(self, "split"):
+                raise ValueError(
+                    "The type of the data is not np.ndarray. "
+                    "You need to define a custom split function "
+                    "named split in your objective class. "
+                    "the type of the function should be "
+                    "(self: Self@Objective, index_train "
+                    "index_test) -> splitted_data"
+                )
+            else:
+                return self.custom_split(*arrays)
+
+        if not hasattr(self, "_cv"):
+            self._cv = itertools.cycle(self.cv.split(*arrays))
```

Comment:
This should go above l.547, so you only do it once in this function, and don't need to do it in `custom_split`.